packages/playwright/src/runner/failureTracker.ts

```diff
@@ -31,8 +31,12 @@ export class FailureTracker {
   }
 
   onTestEnd(test: TestCase, result: TestResult) {
-    if (result.status !== 'skipped' && result.status !== test.expectedStatus)
+    if (result.status === 'skipped')
+      return;
+    if (result.status !== test.expectedStatus)
       ++this._failureCount;
+    else
+      this._failureCount -= result.retry;
```

Comment:
I find it hard to understand this logic. I'd think something like this, similar to `willRetry()` from `base.ts`:

```ts
// Test is considered failing after the last retry.
if (test.outcome() === 'unexpected' && test.results.length > test.retries)
  ++this._failureCount;
```